can/constructor-hydrate/constructor-hydrate.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * @module {connect.Behavior} can-connect/can/constructor-hydrate/constructor-hydrate
+ * @parent can-connect.behaviors
+ *
+ * Persists instances with the same id.
+ *
+ * @signature `constructorHydrate( baseConnection )`
+ *
+ * 	 Overrides [can-define/map/map]'s `setup` method and checks whether a newly created instance already
+ * 	 exists in [can-connect/constructor/store/store.instanceStore]. If it exists that instance will be
+ * 	 returned instead of a new object.
+ *
+ * 	 This behavior has to be used with [can-connect/constructor/store] and [can-connect/can/map/map] behaviors.
+ *
+ * @body
+ *
+ * ## Use
+ *
+ * This behavior could be useful if `Type` converters of [can-define/map/map] are used in multiple places of your app.
```

Comment:
I'm not sure `multiple` totally matters.  

Say I loaded a `User`.  Then I loaded a bunch of `Todo`s, each having a single `owner: User` property.  This would be useful here too.